### ISSUE TITLE
Refactor: implement the managesnapshot.sh features via Java and remove the script

### ISSUE THREAD

**Issue by GutoVeronezi** [apache/cloudstack#6424]
Several snapshot functionalities, in KVM, are implemented via the script `scripts/storage/qcow2/managesnapshot.sh`.  From https://github.com/apache/cloudstack/pull/6422#discussion_r884991838 came the idea to improve these processes by ​​implementing the functionalities via Java, adding documentation, unit tests and logs.

##### ISSUE TYPE
 * Improvement Request

##### COMPONENT NAME
KVM

**Comment by NuxRo** [comment]
Hate to stand in the way of progress, but I believe part of the reason Cloudstack has shell scripts in various places, particularly in dealing with libvirt is because historically the java bindings have lagged behind in functionality etc.

With my Ops hat on, are we sure the shell script bits we want to replace with java are now properly supported in all the major distro, especially the older ones (eg CentOS 7, Ubuntu 20.04) or are we in danger of breaking upgrade paths?
Thoughts @weizhouapache @wido @rohityadavcloud ?

**Comment by wido** [comment]
> Hate to stand in the way of progress, but I believe part of the reason Cloudstack has shell scripts in various places, particularly in dealing with libvirt is because historically the java bindings have lagged behind in functionality etc.
> 
> With my Ops hat on, are we sure the shell script bits we want to replace with java are now properly supported in all the major distro, especially the older ones (eg CentOS 7, Ubuntu 20.04) or are we in danger of breaking upgrade paths? Thoughts @weizhouapache @wido @rohityadavcloud ?

No, I actually think we should implement this in the Java bindings where possible, but ofcourse, caution is needed.

The reason things were implemented in shell scripts is because people didn't bother in implementing it in libvirt-java and thus just took the easier route.

Using the API calls of Libvirt is safer, has better Exception handling and such.

Where possible, use that :-)

**Comment by rohityadavcloud** [comment]
@GutoVeronezi are you working on this?

**Comment by DaanHoogland** [comment]
@GutoVeronezi , given that much work on snapshots has been done since, what is the status of this ticket in your opinion?
